.github/workflows/codeql.yaml

```
@@ -0,0 +1,35 @@
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: ${{ matrix.language }}
+          config-file: ./.github/config/codeql.yaml
+
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v2
```

actionlint (1.7.4)

the runner of "github/codeql-action/analyze@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)